protostar/cheatable_starknet/cheatcodes/cairo_cheatcode.py

```diff
@@ -1,12 +1,65 @@
-from abc import ABC
-from typing import TYPE_CHECKING
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Callable, Union
+
+from starknet_py.cairo.felt import encode_shortstring
 
 from protostar.cairo import HintLocal
+from protostar.cheatable_starknet.cheatable_exception import CheatableException
 
 if TYPE_CHECKING:
     from protostar.cheatable_starknet.cheaters import CairoCheaters
 
+Felt = int
+
+
+@dataclass
+class CheatcodeException(Exception):
+    code: Felt
+    ex: Exception
+
+
+@dataclass
+class CairoCheatcodeInvalidExecution:
+    ok = None
+    err: CheatcodeException
+
+    def unwrap(self) -> Any:
+        raise self.err
+
+
+@dataclass
+class CairoCheatcodeValidExecution:
+    ok: Any
+    err = None
+
+    def unwrap(self) -> Any:
+        return self.ok
+
+
+CairoCheatcodeExecutionResult = Union[
+    CairoCheatcodeValidExecution, CairoCheatcodeInvalidExecution
+]
+
 
 class CairoCheatcode(HintLocal, ABC):
     def __init__(self, cheaters: "CairoCheaters"):
         self.cheaters = cheaters
+
+    @abstractmethod
+    def _build(self) -> Callable:
+        ...
+
+    def build(self):
+        def wrapper(*args: Any, **kwargs: Any):
+            try:
+                result = self._build()(*args, **kwargs)
+                return CairoCheatcodeValidExecution(ok=result)
+            except CheatableException as ex:
```

Comment:
I am not sure about this generalisation, we can leave it for now but it may be necessary to differentiate handling between cheatcodes

Reply:
I don't see a potential use case. If this exception is too concrete, it can be abstracted.